[TestFix] Fix  issues for bulk tests execution

vamahaja opened the pull request:
Fix issues -
- Add timeouts for long executing commands
- Use baseclass function to create large size heketi volume
- Move TC `test_arbiter_scaled_heketi_and_gluster_volume_mapping' to tier4 as it require to run on large no of volumes
- Resource creation get fails due timeout on some cluster configurations. Increase timeout to get resource in `Bound` state even in small configs

vamahaja committed:
[TestFix] Increase timeout for resource creation

Resource creation get fails due timeout on some cluster
configurations. Increase timeout to get resource in `Bound`
state even in small configs.

srivickynesh commented on openshift-storage-libs/openshiftstoragelibs/baseclass.py:
```diff
@@ -1389,3 +1397,109 @@ def verify_setup(self, prefix, timeout=300, wait_step=10):
+
+    def create_heketi_volumes_in_batch(
+            self, heketi_client_node, heketi_server_url, vol_count, size=1,
+            batch_amount=8, timeout=600, interval=5, **kwargs):
```
why is the default timeout value is high, we can have default limit any changes has to come from testcase to increase the limit

vamahaja replied:
This was due to incorrect rebase

srivickynesh commented on openshift-storage-libs/openshiftstoragelibs/baseclass.py:
```diff
@@ -1389,3 +1397,109 @@ def verify_setup(self, prefix, timeout=300, wait_step=10):
 
         self.verify_pods_are_running()
+
+    def create_heketi_volumes_in_batch(
```
There is already an open patch for this method https://github.com/gluster/glusterfs-containers-tests/pull/33

vamahaja replied:
This was due to incorrect rebase

SushilG96 reviewed (changes requested):
Rebase this patch

vamahaja committed:
[TestFix] Move scale TC to tier4

Move TC `test_arbiter_scaled_heketi_and_gluster_volume_mapping' to
tier4 as it require to run on large no of volumes.

vamahaja committed:
[TestFix] Use baseclass function to create large size function

vamahaja committed:
[LiFix] Add timeout for long running commands

SushilG96 commented on openshift-storage-libs/openshiftstoragelibs/heketi_ops.py:
```diff
@@ -2107,7 +2107,8 @@ def validate_dev_path_vg_and_uuid(
 
     # Collect data from the node
-    cmd = "pvs --noheadings -o vg_name,uuid -S name={}".format(dev_name)
+    cmd = ("timeout 10 pvs --noheadings -o vg_name,uuid "
```
Increase the timeout to 120 sec

vamahaja replied:
Done

SushilG96 commented on openshift-storage-libs/openshiftstoragelibs/openshift_storage_libs.py:
```diff
@@ -254,7 +254,8 @@ def get_pvs_info(node, gluster_node_ip, devices_list, raise_on_error=True):
     pvs_info = []
     for device in devices_list:
-        cmd = ("pvs -o pv_name,pv_uuid,vg_name | grep {}".format(device))
+        cmd = ("timeout 10 pvs -o pv_name,pv_uuid,vg_name | "
```
Same as above

vamahaja replied:
Done

adityaramteke commented on tests/functional/heketi/test_block_volumes_heketi.py:
```diff
@@ -310,18 +315,16 @@ def test_create_max_num_blockhostingvolumes(self):
+            bhv_info = self.create_heketi_volume_with_name_and_wait(
+                bhv_name, default_bhv_size,
+                raise_on_cleanup_error=False, json=True, block=True)
+            if bhv_info["id"] not in existing_bhv_list.keys():
+                new_bhv_list.append(bhv_info["id"])
```
```suggestion
            if bhv_info.get("id") not in existing_bhv_list.keys():
                new_bhv_list.append(bhv_info.get("id"))
```

vamahaja replied:
Please check PR title, it's meant to fix issues related logic. Suggested change won't affect any steps/logic which will fix the issue.

adityaramteke replied:
Yes, do you mean suggested conventional changes are not valid with this PR?

vamahaja replied:
What do you mean by conventional ? We switched to use `get` for dict in sametime back long time after this TC. And it's not mandatory use `get`. So this won't affect the code.

adityaramteke replied:
Sure, Vaibhav! It is conventional as `.get` can be used to avoid KeyError. Well, it does not fix the original logic.

adityaramteke commented on tests/functional/heketi/test_block_volumes_heketi.py:
```diff
@@ -310,18 +315,16 @@ def test_create_max_num_blockhostingvolumes(self):
+            if bhv_info["id"] not in existing_bhv_list.keys():
+                new_bhv_list.append(bhv_info["id"])
+
+            free_size = bhv_info["blockinfo"]["freesize"]
```
```suggestion
            free_size = bhv_info.get("blockinfo").get("freesize")
```

vamahaja replied:
same as above
